mods/ra/maps/allies-06a/allies06a.lua

```diff
@@ -0,0 +1,215 @@
+	{ TnkPatrol8.Location, TnkPatrol1.Location, TnkPatrol2.Location, TnkPatrol3.Location, TnkPatrol4.Location, TnkPatrol5.Location,  TnkPatrol6.Location, TnkPatrol7.Location }
+}
+
+SovietSubPath = { SubPatrol3_1.Location, SubPatrol3_2.Location, SubPatrol3_3.Location }
+
+ParadropWaypoints =
+{
+	easy = { UnitBStopLocation },
+	normal = { UnitBStopLocation, UnitAStopLocation },
+	hard = { UnitBStopLocation, MCVStopLocation, UnitAStopLocation }
+}
+
+SovietTechLabs = { TechLab1, TechLab2, TechLab3 }
+TechLabCams = { TechCam1, TechCam2, TechCam3 }
+
+GroupPatrol = function(units, waypoints, delay)
+	local i = 1
+	local stop = false
+
+	Utils.Do(units, function(unit)
+		Trigger.OnIdle(unit, function()
+			if stop then
+				return
+			end
+			if unit.Location == waypoints[i] then
+				local bool = Utils.All(units, function(actor) return actor.IsIdle end)
+				if bool then
+					stop = true
+					i = i + 1
+					if i > #waypoints then
+						i = 1
+					end
+					Trigger.AfterDelay(delay, function() stop = false end)
+				end
+			else
+				unit.AttackMove(waypoints[i])
+			end
+		end)
+	end)
+end
+
+InitialSovietPatrols = function()
+	-- Dog Patrols
+	for i = 1, 4 do
+		GroupPatrol(SovietDogPatrols[i], SovietDogPatrolPaths[i], DateTime.Seconds(5))
+	end
+
+	-- Mammoth Patrols
+	for i = 1, 3 do
+		Trigger.AfterDelay(DateTime.Seconds(6 * (i - 1)), function()
+			Trigger.OnIdle(Mammoths[i], function()
+				Mammoths[i].Patrol(SovietMammothPaths[i])
+			end)
+		end)
+	end
+
+	-- Sub Patrols
+	Patrol1Sub.Patrol({ SubPatrol1_1.Location, SubPatrol1_2.Location })
+	Patrol2Sub.Patrol({ SubPatrol2_1.Location, SubPatrol2_2.Location })
+	Patrol3Sub1.Patrol(SovietSubPath)
+	Patrol3Sub2.Patrol(SovietSubPath)
+end
+
+InitialAlliedReinforcements = function()
+	local camera = Actor.Create("Camera", true, { Owner = player, Location = DefaultCameraPosition.Location })
+	Trigger.AfterDelay(DateTime.Seconds(30), camera.Destroy)
+
+	Reinforcements.Reinforce(player, AlliedReinforcementsA, { AlliedEntry1.Location, UnitBStopLocation.Location }, 2)
+	Trigger.AfterDelay(DateTime.Seconds(2), function()
+		Reinforcements.Reinforce(player, AlliedReinforcementsB, { AlliedEntry2.Location, UnitAStopLocation.Location }, 2)
+	end)
+	Trigger.AfterDelay(DateTime.Seconds(5), function()
+		Reinforcements.Reinforce(player, { "mcv" }, { AlliedEntry3.Location, MCVStopLocation.Location })
+	end)
+end
+
+CaptureRadarDome = function()
+	Trigger.OnKilled(Radar, function()
+		player.MarkFailedObjective(CaptureRadarDomeObj)
+	end)
+
+	Trigger.OnCapture(Radar, function()
+		player.MarkCompletedObjective(CaptureRadarDomeObj)
+	end)
+end
+
+InfiltrateTechCenter = function()
+	Utils.Do(SovietTechLabs, function(a)
+		Trigger.OnInfiltrated(a, function()
+			DestroySovietsObj = player.AddPrimaryObjective("Destroy all Soviet buildings and units in the area.")
```

Comment:
I think there should be a check here whether another tech lab has already been infiltrated, otherwise you'd get duplicate objectives, and you'd be overwriting the reference to the earlier one(s) so you cannot mark them complete anymore.